client/src/utils/date.js

```diff
@@ -1,3 +1,28 @@
 export const getDateString = (date) => {
     return new Date(date).toISOString().split('T')[0]
-}
\ No newline at end of file
+}
+
+export const getDatesOfPastMonth = () => {
+    const currentDate = new Date();
+    const currentMonth = currentDate.getMonth() + 1;
```

Comment:
small nitpick, but it seems getMonth is 0-indexed (0 = January, 11 = December) and I think if you remove the +1 here, you can remove the -1 at line 11, so in the end you'll only have a +1 on line 13 and 20, so there's less magic numbers! It's just a bit hard to keep up with all the magic numbers, so it might make more sense to keep it 0-indexed and only add +1 when needed. I was confused why you added 1 to currentMonth at first here.

Reply:
Actually now that I think about it, is it off by 1 month? If today was March 4, currentDate.getMonth() would return 2, so currentMonth = 3 because of the +1, startDate's month is currentMonth-1 which is 2, end date's month is currentMonth which is 3.  currentDateIterator takes the startDate, so getMonth returns 2.  The String you create for month would be 03 due to the padding. But if today was March 4 and we want the dates of the past month, we would want every date starting from February 4 to March 4 right?

Reply:
king thanks for catching that I used the wrong function, I used `getDay` instead of `getDate`!

Reply:
I changed it and I think it's using the correct start and end dates now!
<img width="1115" alt="image" src="https://github.com/phillytan/FinanceTracker/assets/64379928/ee0aabd5-41c9-4566-ae29-da0bdb90b991">